engine/src/test/java/io/camunda/zeebe/process/test/engine/EngineClientTest.java

```diff
@@ -1231,6 +1237,45 @@ void shouldBroadcastSignal() {
     assertThat(response.getKey()).isPositive();
   }
 
+  @Test
+  public void shouldNotSendResponseWhenThereNoCorrespondingRequest() {
+    // given
+    zeebeClient
+        .newDeployResourceCommand()
+        .addProcessModel(
+            Bpmn.createExecutableProcess("process")
+                .startEvent()
+                .intermediateThrowEvent("signal", b -> b.signal("signalName"))
+                .endEvent()
+                .done(),
+            "process.bpmn")
+        .send()
+        .join();
+
+    // when
+    zeebeClient.newCreateInstanceCommand().bpmnProcessId("process").latestVersion().send().join();
+
+    // then
+    Awaitility.await()
+        .untilAsserted(
+            () -> {
+              final Optional<Record<ProcessInstanceRecordValue>> processCompleted =
+                  StreamSupport.stream(
+                          RecordStream.of(zeebeEngine.getRecordStreamSource())
+                              .processInstanceRecords()
+                              .spliterator(),
+                          false)
+                      .filter(r -> r.getValue().getBpmnElementType() == BpmnElementType.PROCESS)
+                      .filter(r -> r.getValue().getBpmnEventType() == BpmnEventType.UNSPECIFIED)
+                      .filter(r -> r.getIntent() == ProcessInstanceIntent.ELEMENT_COMPLETED)
+                      .findFirst();
+
+              assertThat(processCompleted).isNotEmpty();
+            });
+
+    assertFalse(responseIntents.contains(SignalIntent.BROADCASTED));
```

Comment:
🔧 Let's not mix JUnit asserts and AssertJ. Rather use `assertThat(...).isFalse()` or assert the list directly, e.g. `assertThat(responseIntents).doesNotContain(SignalIntent.BROADCASTED)`.